### DISCUSSION TITLE
Suggestion: insert with gravity

### DISCUSSION THREAD

**Discussion by jfcalvo** [libvips/libvips#2060] [Ideas]
I have been using `gravity` lately a lot (super useful the `direction` parameter) but now that I need to insert an image into  another one I need to calculate the `x` and `y` positions using `insert` operation.

Why not to have an `insert` operation that doesn't receive `x` and `y` and instead of that receive a `direction` parameter as `gravity` has?

I think this would be super useful in a lot of cases when you need to insert an image into the other but you don't want to implement all the logic to calculate `x` and `y` parameters.

Thank you very much for all the help.

**Comment by jcupitt** [comment]
Hello @jfcalvo,

I did think about something like this, but it's not quite as useful as you'd think (in my experience). For example, you often want the smaller image placed with a small margin (eg. 100 pixels in from the right edge), and once you start adding horizontal and vertical margins too, the API becomes complex and annoying.

Perhaps it would be worthwhile though? Let's leave this tagged as an enhancement.

**Comment by jfcalvo** [comment]
Thanks for the quick response @jcupitt 

I'm not sure what are you trying to mean with the margins. I'm only talking about implementing something equal to `insert` operation but instead of receiving `x` and `y` using a `direction` parameter. The same that is defined on `gravity`.

Why is that adding a lot of complexity to the API? Is not possible to reuse the same logic that `gravity` is using?
